akka-sample-sharding-scala/src/main/scala/sample/sharding/WeatherRoutes.scala

```diff
@@ -0,0 +1,44 @@
+package sample.sharding
+
+import java.util.concurrent.TimeUnit
+
+import scala.util.{ Failure, Success }
+import akka.actor.typed.{ ActorRef, ActorSystem }
+import akka.{ actor => classic }
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Route
+import akka.util.Timeout
+
+class WeatherRoutes(guardian: ActorRef[Guardian.Command])(implicit system: ActorSystem[_]) {
+
+  import akka.actor.typed.scaladsl.adapter._
+  implicit val classicSystem: classic.ActorSystem = system.toClassic
+
+  implicit val timeout: Timeout =
+    Timeout(system.settings.config.getDuration("weather.routes.ask-timeout").toMillis, TimeUnit.MILLISECONDS)
```

Comment:
Annoying that it is less code in Java (`akka.util.JavaDurationConverters` is Akka internal) :P

Reply:
https://github.com/akka/akka/issues/28105

Reply:
a nice trick I saw somewhere else is to use the java api:
```
implicit val timeout: Timeout =
    Timeout.create(system.settings.config.getDuration("weather.routes.ask-timeout"))
```

Reply:
Yeah, and for some reason the way I used to do it fails in compile

```scala
import akka.util.Helpers.ConfigOps
  implicit val timeout: Timeout =
    Timeout(system.settings.config.getMillisDuration("killrweather.routes.ask-timeout"))
```

`value getMillisDuration is not a member of com.typesafe.config.Config`

Reply:
```scala
import scala.concurrent.duration._
implicit val timeout: Timeout = system.settings.config.getDuration("key").toMillis.millis
```